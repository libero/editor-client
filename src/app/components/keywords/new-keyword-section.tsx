import { EditorState, Transaction } from 'prosemirror-state';
import React from 'react';
import { EditorView } from 'prosemirror-view';
import { ProseMirrorEditorView } from 'app/components/rich-text-editor/prosemirror-editor-view';

const ENTER_KEY_CODE = 'Enter';

interface NewKeywordSection {
  className?: string;
  editorState: EditorState;
  onEnter: (editorState: EditorState) => void;
  onChange: (change: Transaction) => void;
  onFocus: () => void;
  onBlur: () => void;
}

export const NewKeywordSection: React.FC<NewKeywordSection> = ({
  editorState,
  onEnter,
  onChange,
  className,
  onFocus,
  onBlur
}) => {
  const options = {
    handleDOMEvents: {
      focus: () => {
        onFocus();
        return false;
      },
      blur: () => {
        onBlur();
        return false;
      }
    },
    handleKeyDown: (view: EditorView, event: KeyboardEvent): boolean => {
      if (event.key === ENTER_KEY_CODE) {
        onEnter(view.state);
        return true;
      }
      return false;
    }
  };

  return (
    <ProseMirrorEditorView className={className} options={options} editorState={editorState} onChange={onChange} />
  );
};
