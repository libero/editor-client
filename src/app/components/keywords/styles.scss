.keywords-group-container {

  .keyword-group-legend {
    font-size: 12px;
    color: #263238;
  }

  .keywords-section {
    display: flex;
    flex-flow: wrap;
    font-size: 13px;
    line-height: 32px;
    cursor: text;

    .new-keyword-container {
      flex: 1;

      .ProseMirror-focused {
        outline: none;
      }
    }

    .keyword {
      display: flex;
      background: #087ACC;
      color: #fff;
      font-size: 13px;
      line-height: normal;
      white-space: nowrap;
      margin: 2px 4px;
      padding: 6px 6px 6px 12px;
      border-radius: 16px;

      &.focused {
        background: white;
        color: #263238;
      }

      .content-container {
        flex: 1;
      }

      .ProseMirror-selectednode, .ProseMirror-focused {
        outline: none;
        border-color: #8cf;
      }

      .keyword-delete-cta {
        width: 20px;
        height: 20px;
        margin-left: 8px;
      }
    }
  }
}