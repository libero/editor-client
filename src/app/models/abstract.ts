import { EditorState } from 'prosemirror-state';
import { DOMParser as ProseMirrorDOMParser } from 'prosemirror-model';
import { gapCursor } from 'prosemirror-gapcursor';

import { makeSchemaFromConfig } from 'app/models/utils';
import * as abstractConfig from 'app/models/config/abstract.config';
import { buildInputRules } from 'app/models/plugins/input-rules';
import { PlaceholderPlugin } from 'app/models/plugins/placeholder.plugin';
import { SelectionPlugin } from 'app/models/plugins/selection.plugins';

export function createAbstractState(content: Element): EditorState {
  const schema = makeSchemaFromConfig(abstractConfig.topNode, abstractConfig.nodes, abstractConfig.marks);
  const xmlContentDocument = document.implementation.createDocument('', '', null);

  if (content) {
    xmlContentDocument.appendChild(content);
  }

  return EditorState.create({
    doc: ProseMirrorDOMParser.fromSchema(schema).parse(xmlContentDocument),
    schema,
    plugins: [buildInputRules(), SelectionPlugin, gapCursor(), PlaceholderPlugin('Enter abstract')]
  });
}

export function createImpactStatementState(content: Element): EditorState {
  const schema = makeSchemaFromConfig(abstractConfig.topNode, abstractConfig.nodes, abstractConfig.marks);
  const xmlContentDocument = document.implementation.createDocument('', '', null);

  if (content) {
    xmlContentDocument.appendChild(content);
  }

  return EditorState.create({
    doc: ProseMirrorDOMParser.fromSchema(schema).parse(xmlContentDocument),
    schema,
    plugins: [buildInputRules(), SelectionPlugin, gapCursor(), PlaceholderPlugin('Enter impact statement')]
  });
}
