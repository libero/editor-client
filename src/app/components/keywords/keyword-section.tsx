import { EditorState, Transaction } from 'prosemirror-state';
import React, { useCallback, useEffect, useRef } from 'react';
import classNames from 'classnames';
import { EditorView } from 'prosemirror-view';
import { IconButton } from '@material-ui/core';
import CancelIcon from '@material-ui/icons/Cancel';

import { ProseMirrorEditorView } from 'app/components/rich-text-editor/prosemirror-editor-view';
import { useKeywordStyles } from './styles';
import { Keyword } from 'app/types/manuscript';

const ENTER_KEY_CODE = 'Enter';

interface KeywordProps {
  keyword: Keyword;
  onChange: (id: string, state: Transaction) => void;
  onDelete: (keyword: Keyword) => void;
  isActive: boolean;
  onFocus: (state: EditorState, id: string) => void;
  onBlur: () => void;
}

export const KeywordSection: React.FC<KeywordProps> = (props) => {
  const { keyword, onDelete, onChange, onFocus, isActive, onBlur } = props;
  const prosemirrorRef = useRef(null);
  const containerRef = useRef<HTMLDivElement>(null);
  const deleteButtonRef = useRef(null);
  const classes = useKeywordStyles();

  const focusOnDblClick = useCallback(() => {
    prosemirrorRef.current.focus();
  }, []);

  const preventSingleClickWhenInactive = useCallback(
    (event: Event) => {
      const isDeleteButtonClick = deleteButtonRef.current.contains(event.target);
      if (!isActive) {
        event.stopPropagation();
        event.preventDefault();
      }

      if (isDeleteButtonClick) {
        onDelete(keyword);
        onBlur();
      }
    },
    [isActive, onDelete, keyword, onBlur]
  );

  useEffect(() => {
    const containerEl = containerRef.current;
    if (containerEl) {
      containerEl.addEventListener('dblclick', focusOnDblClick, true);
      containerEl.addEventListener('mousedown', preventSingleClickWhenInactive, true);
    }

    return () => {
      containerEl.removeEventListener('dblclick', focusOnDblClick, true);
      containerEl.removeEventListener('mousedown', preventSingleClickWhenInactive, true);
    };
  }, [containerRef, focusOnDblClick, preventSingleClickWhenInactive]);

  const handleFocusEvent = (view: EditorView): boolean => {
    onFocus(view.state, keyword.id);
    return true;
  };

  const options = {
    handleDOMEvents: {
      focus: handleFocusEvent,
      keydown: (view: EditorView, event: KeyboardEvent): boolean => {
        if (event.key === ENTER_KEY_CODE) {
          prosemirrorRef.current.blur();
          onBlur();
          return true;
        }
        return false;
      }
    }
  };

  const handleChange = useCallback(
    (transaction: Transaction) => {
      onChange(keyword.id, transaction);
    },
    [onChange, keyword.id]
  );

  return (
    <div
      className={classNames(classes.keyword, { focused: isActive })}
      ref={containerRef}
      data-test-id="keyword-container"
    >
      <ProseMirrorEditorView
        options={options}
        ref={prosemirrorRef}
        editorState={keyword.content}
        onChange={handleChange}
      />
      <IconButton
        ref={deleteButtonRef}
        aria-label="delete keyword"
        tabIndex={0}
        color="primary"
        size="small"
        className={isActive ? classes.hidden : ''}
      >
        <CancelIcon className={classes.deleteIcon} />
      </IconButton>
    </div>
  );
};
