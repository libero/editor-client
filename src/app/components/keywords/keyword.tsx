import { EditorState, Transaction } from 'prosemirror-state';
import React, { useCallback, useEffect, useRef } from 'react';
import classNames from 'classnames';
import { EditorView } from 'prosemirror-view';
import { IconButton } from '@material-ui/core';
import CancelIcon from '@material-ui/icons/Cancel';

import { ProseMirrorEditorView } from 'app/components/rich-text-editor/prosemirror-editor-view';
import { useKeywordStyles } from './styles';

const ENTER_KEY_CODE = 'Enter';

interface KeywordProps {
  editorState: EditorState;
  onChange: (state: Transaction) => void;
  onDelete: () => void;
  isActive: boolean;
  onFocus: (state: EditorState) => void;
  onBlur: () => void;
}

export const Keyword: React.FC<KeywordProps> = (props) => {
  const { editorState, onDelete, onChange, onFocus, isActive, onBlur } = props;
  const prosemirrorRef = useRef(null);
  const containerRef = useRef<HTMLDivElement>(null);
  const deleteButtonRef = useRef(null);
  const classes = useKeywordStyles();

  const focusOnDblClick = useCallback(() => {
    prosemirrorRef.current.focus();
  }, []);

  const preventSingleClickWhenInactive = useCallback(
    (event: Event) => {
      const isDeleteButtonClick = deleteButtonRef.current.contains(event.target);
      if (!isActive) {
        event.stopPropagation();
        event.preventDefault();
      }

      if (isDeleteButtonClick) {
        onDelete();
      }
    },
    [isActive, onDelete]
  );

  useEffect(() => {
    const containerEl = containerRef.current;
    if (containerEl) {
      containerEl.addEventListener('dblclick', focusOnDblClick, true);
      containerEl.addEventListener('mousedown', preventSingleClickWhenInactive, true);
    }

    return () => {
      containerEl.removeEventListener('dblclick', focusOnDblClick, true);
      containerEl.removeEventListener('mousedown', preventSingleClickWhenInactive, true);
    };
  }, [containerRef, focusOnDblClick, preventSingleClickWhenInactive]);

  const handleFocusEvent = (view: EditorView): boolean => {
    onFocus(view.state);
    return true;
  };

  const options = {
    handleDOMEvents: {
      focus: handleFocusEvent,
      keydown: (view: EditorView, event: KeyboardEvent): boolean => {
        if (event.key === ENTER_KEY_CODE) {
          prosemirrorRef.current.blur();
          onBlur();
          return true;
        }
        return false;
      }
    }
  };

  return (
    <div
      className={classNames(classes.keyword, { focused: isActive })}
      ref={containerRef}
      data-test-id="keyword-container"
    >
      <ProseMirrorEditorView options={options} ref={prosemirrorRef} editorState={editorState} onChange={onChange} />
      <IconButton
        ref={deleteButtonRef}
        aria-label="delete keyword"
        tabIndex={0}
        color="primary"
        size="small"
        className={isActive ? classes.hidden : ''}
      >
        <CancelIcon className={classes.deleteIcon} />
      </IconButton>
    </div>
  );
};
