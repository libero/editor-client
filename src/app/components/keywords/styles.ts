import { makeStyles } from '@material-ui/core/styles';
import { fade } from '@material-ui/core';

export const makeKeywordContainerStyles = makeStyles((theme) => ({
  keywordsEditor: {
    padding: theme.spacing(0.5, 1, 1, 1),
    border: `solid 1px`,
    borderColor: theme.palette.secondary.dark,
    borderRadius: theme.spacing(0.5),
    boxSizing: 'border-box',
    '&:focus': {
      outline: 'none',
      borderColor: theme.palette.primary.light,
      '& > legend': {
        color: theme.palette.primary.main
      }
    }
  },

  keywordsSection: {
    display: 'flex',
    flexFlow: 'wrap',
    cursor: 'text'
  },

  newKeywordEditor: {
    flex: 1,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: theme.spacing(0.5, 0.5, 0.5, 1.5),
    '& .ProseMirror': {
      flex: 1
    }
  }
}));

export const useKeywordStyles = makeStyles((theme) => ({
  keyword: {
    display: 'flex',
    alignItems: 'center',
    background: theme.palette.primary.main,
    whiteSpace: 'nowrap',
    margin: theme.spacing(0.25, 1, 0.25, 0),
    fontSize: '0.9rem',
    padding: theme.spacing(0.5, 0.5, 0.5, 1.5),
    borderRadius: 16,

    '& .ProseMirror': {
      color: theme.palette.primary.contrastText
    },

    '&.focused': {
      background: 'transparent',
      '& .ProseMirror': {
        color: theme.palette.text.primary
      }
    }
  },

  deleteIcon: {
    WebkitTapHighlightColor: 'transparent',
    color: fade(theme.palette.primary.contrastText, 0.7),
    height: 20,
    width: 20,
    cursor: 'pointer',
    '&:hover, &:active, &:focus': {
      color: theme.palette.primary.contrastText
    }
  }
}));
