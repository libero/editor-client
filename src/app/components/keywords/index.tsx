import React, { useCallback } from 'react';
import { EditorState, Transaction } from 'prosemirror-state';

import { NewKeywordSection } from './new-keyword-section';
import { Keyword } from './keyword';
import { makeKeywordContainerStyles } from './styles';
import { SectionContainer } from 'app/components/section-container';

interface KeywordsEditorProps {
  keywords: EditorState[];
  newKeyword: EditorState;
  label?: string;
  name: string;
  activeKeywordPath?: string;
  onDelete: (groupType: string, index: number) => void;
  onChange: (groupType: string, index: number, change: Transaction) => void;
  onNewKeywordChange: (groupType: string, change: Transaction) => void;
  onAdd: (groupType: string, state: EditorState) => void;
  onFocusSwitch: (groupType: string, index: number | undefined, isNewKeywordFocused?: boolean) => void;
  onBlur: () => void;
}

const isActiveKeyword = (group: string, index: number, activeKeywordPath?: string) => {
  return activeKeywordPath === `keywordGroups.${group}.keywords.${index}`;
};

export const KeywordsEditor: React.FC<KeywordsEditorProps> = (props) => {
  const {
    keywords,
    label,
    onChange,
    onDelete,
    name,
    onAdd,
    onFocusSwitch,
    newKeyword,
    onNewKeywordChange,
    activeKeywordPath,
    onBlur
  } = props;
  const classes = makeKeywordContainerStyles();

  const handleFocus = useCallback(
    (index: number) => {
      onFocusSwitch(name, index);
    },
    [name, onFocusSwitch]
  );

  const renderKeywords = (keywords: EditorState[]) => {
    return keywords.map((keywordEditorState, index) => {
      return (
        <Keyword
          key={index}
          isActive={isActiveKeyword(name, index, activeKeywordPath)}
          onChange={onChange.bind(null, name, index)}
          editorState={keywordEditorState}
          onDelete={onDelete.bind(null, name, index)}
          onFocusSwitch={handleFocus.bind(null, index)}
          onBlur={onBlur}
        />
      );
    });
  };

  const handleNewKeywordFocus = useCallback(() => {
    onFocusSwitch(name, undefined, true);
  }, [name, onFocusSwitch]);

  const isGroupFocused = activeKeywordPath && activeKeywordPath.startsWith(`keywordGroups.${name}`);

  return (
    <SectionContainer label={label} focused={isGroupFocused}>
      <section className={classes.keywordsSection}>
        {renderKeywords(keywords)}
        <NewKeywordSection
          className={classes.newKeywordEditor}
          onChange={onNewKeywordChange.bind(null, name)}
          editorState={newKeyword}
          onEnter={onAdd.bind(null, name)}
          onFocus={handleNewKeywordFocus}
        />
      </section>
    </SectionContainer>
  );
};
