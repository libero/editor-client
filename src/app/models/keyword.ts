import { DOMParser as ProseMirrorDOMParser } from 'prosemirror-model';
import { EditorState } from 'prosemirror-state';
import { gapCursor } from 'prosemirror-gapcursor';
import { dropCursor } from 'prosemirror-dropcursor';
import { keymap } from 'prosemirror-keymap';
import { baseKeymap } from 'prosemirror-commands';

import { makeSchemaFromConfig } from 'app/models/utils';
import * as keywordConfig from 'app/models/config/keywords.config';
import { buildInputRules } from 'app/models/plugins/input-rules';
import { BackmatterEntity } from 'app/models/backmatter-entity';
import { JSONObject } from 'app/types/utility.types';
import { SelectionPlugin } from 'app/models/plugins/selection.plugins';

export class Keyword extends BackmatterEntity {
  content: EditorState;

  constructor(data?: JSONObject | Element | EditorState) {
    super();
    if (data instanceof EditorState) {
      this.content = data;
    } else {
      this.createEntity(data);
    }
  }

  public clone(): Keyword {
    const kwd = new Keyword();
    kwd._id = this.id;
    kwd.content = this.content.apply(this.content.tr);
    return kwd;
  }

  protected fromXML(xmlNode: Element): void {
    const schema = makeSchemaFromConfig(keywordConfig.topNode, keywordConfig.nodes, keywordConfig.marks);
    this.content = EditorState.create({
      doc: ProseMirrorDOMParser.fromSchema(schema).parse(xmlNode),
      schema,
      plugins: [buildInputRules(), SelectionPlugin, gapCursor(), dropCursor(), keymap(baseKeymap)]
    });
  }

  protected fromJSON(json: JSONObject): void {
    this._id = (json._id as string) || this._id;
    const blankState = this.createEmptyEditorState();
    this.content = EditorState.fromJSON(
      { schema: blankState.schema, plugins: blankState.schema.plugins },
      json.content as JSONObject
    );
  }

  protected createBlank(): void {
    this.content = this.createEmptyEditorState();
  }

  private createEmptyEditorState(): EditorState {
    const schema = makeSchemaFromConfig(keywordConfig.topNode, keywordConfig.nodes, keywordConfig.marks);
    return EditorState.create({
      doc: undefined,
      schema,
      plugins: [buildInputRules(), SelectionPlugin, gapCursor(), dropCursor(), keymap(baseKeymap)]
    });
  }
}

export interface KeywordGroup {
  title: string | undefined;
  keywords: Keyword[];
  newKeyword: Keyword;
}

export interface KeywordGroups {
  [keywordType: string]: KeywordGroup;
}

export function createKeywordGroupsState(keywordGroupsXml: Element[]): KeywordGroups {
  return keywordGroupsXml.reduce((acc, kwdGroup) => {
    const kwdGroupType = kwdGroup.getAttribute('kwd-group-type') || 'keywords-1';
    const groupTitle = kwdGroup.querySelector('title');
    const moreKeywords = Array.from(kwdGroup.querySelectorAll('kwd')).map(
      (keywordEl: Element, _) => new Keyword(keywordEl)
    );
    acc[kwdGroupType] = {
      title: groupTitle ? groupTitle.textContent : undefined,
      keywords: moreKeywords,
      newKeyword: new Keyword()
    };

    return acc;
  }, {});
}
