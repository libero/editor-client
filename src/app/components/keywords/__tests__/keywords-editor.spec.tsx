import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import { KeywordsEditor } from 'app/components/keywords/index';
import { KeywordSection } from 'app/components/keywords/keyword-section';
import { NewKeywordSection } from 'app/components/keywords/new-keyword-section';
import { createNewKeywordState } from 'app/models/keyword';

describe('KeywordsEditorComponent', () => {
  it('renders keywords', () => {
    const props = {
      keywords: [createNewKeywordState()],
      newKeyword: createNewKeywordState(),
      name: 'keyword-group',
      label: 'Keywords label',
      onDelete: jest.fn(),
      onChange: jest.fn(),
      onAdd: jest.fn(),
      onFocus: jest.fn(),
      onNewKeywordChange: jest.fn(),
      onBlur: jest.fn()
    };

    const component = create(<KeywordsEditor {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('fires events when changes appear on an existing keyword', () => {
    const props = {
      keywords: [createNewKeywordState()],
      newKeyword: createNewKeywordState(),
      name: 'keyword-group',
      label: 'Keywords label',
      onDelete: jest.fn(),
      onChange: jest.fn(),
      onAdd: jest.fn(),
      onFocus: jest.fn(),
      onNewKeywordChange: jest.fn(),
      onBlur: jest.fn()
    };

    const component = shallow(<KeywordsEditor {...props} />);
    const keywordProps = component.find(KeywordSection).props();
    const change = props.keywords[0].content.tr;

    keywordProps.onChange(props.keywords[0].id, change);
    expect(props.onChange).toBeCalledWith(props.name, props.keywords[0].id, change);

    keywordProps.onDelete(props.keywords[0]);
    expect(props.onDelete).toBeCalledWith(props.name, props.keywords[0]);
  });

  it('fires events when changes appear on an a new keyword', () => {
    const props = {
      keywords: [createNewKeywordState()],
      newKeyword: createNewKeywordState(),
      name: 'keyword-group',
      label: 'Keywords label',
      onDelete: jest.fn(),
      onChange: jest.fn(),
      onAdd: jest.fn(),
      onFocus: jest.fn(),
      onNewKeywordChange: jest.fn(),
      onBlur: jest.fn()
    };

    const component = shallow(<KeywordsEditor {...props} />);
    const newKeywordProps = component.find(NewKeywordSection).props();
    newKeywordProps.onEnter(newKeywordProps.editorState);
    expect(props.onAdd).toBeCalledWith(props.name, { id: props.newKeyword.id, content: newKeywordProps.editorState });
  });
});
