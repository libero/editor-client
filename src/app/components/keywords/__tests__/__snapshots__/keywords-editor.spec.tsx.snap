// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordsEditorComponent renders keywords 1`] = `
<div
  className="makeStyles-root-4 makeStyles-plainVariant-6"
>
  <label
    className="makeStyles-label-7"
  >
    Keywords label
  </label>
  <section
    className="makeStyles-keywordsSection-2"
  >
    <div
      className="makeStyles-keyword-9"
      data-test-id="keyword-container"
    >
      <div
        className="prosemirrorContainer"
      />
      <button
        aria-label="delete keyword"
        className="MuiButtonBase-root MuiIconButton-root keyword-delete-cta MuiIconButton-colorPrimary MuiIconButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-deleteIcon-10"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      className="prosemirrorContainer makeStyles-newKeywordEditor-3"
    />
  </section>
</div>
`;
